### DISCUSSION TITLE
underscore in subdomain

### DISCUSSION THREAD

**Discussion by cr-sh** [vercel/community#2400] [Bug Report]
Hi, I need to plug a subdomain starting with `_` to vercel, but it seems that it's forbidden. Can I know why?

Domains with underscores are very common in the wild. Check `_jabber._tcp.gmail.com` or `_sip._udp.apnic.net`. I've several domains like `_.domain.tld` on cloudflare, and they work perfectly, even with valid SSL certificates.

In fact, it's perfectly legal to have an underscore in a domain name. Let me quote the standard, RFC 2181, section 11, "Name syntax":

The DNS itself places only one restriction on the particular labels that can be used to identify resource records. That one restriction relates to the length of the label and the full name. [...] Implementations of the DNS protocols must not place any restrictions on the labels that can be used. In particular, DNS servers must not refuse to serve a zone because it contains labels that might not be acceptable to some DNS client programs.

See also the original DNS specification, RFC 1034, section 3.5 "Preferred name syntax" but read it carefully.

Cheers

**Comment by smaeda-ks** [comment]
Hi @cr-sh, the main reason is we use Let's Encrypt for our auto-certificate issuance, and they don't allow underscores in its SAN domains. Therefore, we don't allow it (plus, Vercel enforces HTTPS by default). You might be able to work around it by issuing a wildcard certificate. Have you tried that (adding a relevant wildcard domain to the project instead)?

**Reply by smaeda-ks** [reply]
https://cabforum.org/2018/11/12/ballot-sc-12-sunset-of-underscores-in-dnsnames/

**Comment by cr-sh** [comment]
Thanks for the reply @smaeda-ks :)
